src/request/components/request-detail-panel-logging.tsx

```diff
@@ -2,10 +2,71 @@
 
 import { FontAwesomeIcon } from '../../shell/components/FontAwesomeIcon';
 import { ILogMessage } from '../messages/ILogMessage';
-import { ILoggingComponentModel, ILoggingLevelModel } from '../component-models/ILoggingComponentModel';
+import { ILoggingComponentModel, ILoggingLevelModel, ILogMessageModel } from '../component-models/ILoggingComponentModel';
 
 import _ = require('lodash');
 import React = require('react');
+import Highlight = require('react-highlight');
+
+interface ILogMessageProps {
+    message: ILogMessageModel;
+}
+
+interface ILogMessageState {
+    isExpanded: boolean;
+}
+
+class LogMessage extends React.Component<ILogMessageProps, ILogMessageState> {
+    constructor(props?) {
+        super(props);
+
+        this.state = {
+            isExpanded: false
+        };
+    }
+
+    public render() {
+        return (
+            <div className='tab-logs-table-message'>
+                <div className={this.getIconClass()} onClick={e => this.onToggleExpansion()}><FontAwesomeIcon path={this.getIconPath()} /></div>
+                {
+                    (this.state.isExpanded && this.props.message.isObject)
+                        ? <div className='tab-logs-table-message-object'><Highlight language='javascript'>{this.props.message.message}</Highlight></div>
+                        : <div className={this.getMessageClass()}>{this.props.message.spans.map(span => <span className={span.wasReplaced ? 'tab-logs-table-message-replaced-region' : ''}>{span.text}</span>)}</div>
+                }
```

Comment:
Yes, the `getMessageClass()` returns a truncated/wrapping style based on the expansion state.